pkgs/development/compilers/atasm/default.nix

```diff
@@ -0,0 +1,64 @@
+{ lib
+, stdenv
+, fetchurl
+, unzip
+, zlib
+}:
+
+stdenv.mkDerivation rec {
+  pname = "atasm";
+  version = "1.09";
+
+  src = fetchurl {
+    url = "https://atari.miribilist.com/${pname}/${pname}${builtins.replaceStrings ["."] [""] version}.zip";
+    hash = "sha256-26shhw2r30GZIPz6S1rf6dOLKRpgpLwrqCRZX3+8PvA=";
+  };
+
+  patches = [
+    # make install fails because atasm.txt was moved; report to upstream
+    ./0000-file-not-found.diff
+    # select flags for compilation
+    ./0001-select-flags.diff
+  ];
+
+  dontConfigure = true;
+
+  nativeBuildInputs = [
+    unzip
+  ];
+
+  buildInputs = [
+    zlib
+  ];
+
+  preBuild = ''
+    makeFlagsArray+=(
```

Comment:
Use makeFlags instead.

Reply:
What about the whitespaces?

Reply:
What do you mean? The quoting in makeFlags is not suprt important and in the end it is bash things anyway.

Reply:
How to write this in `makeFlags`?

`USEZ="-DZLIB_CAPABLE -I${zlib}/include"`

Reply:
```nix
makeFlags = [ "USEZ=\"-DZLIB_CAPABLE -I${zlib}/include\"" ]
```
or
```nix
makeFlags = [ ''USEZ="-DZLIB_CAPABLE -I${zlib}/include"'' ]
```

Reply:
```
       > build flags: SHELL=/nix/store/dpjnjrqbgbm8a5wvi1hya01vd8wyvsq4-bash-4.4-p23/bin/bash -C ./src CC=cc USEZ=\"-DZLIB_CAPABLE -I/nix/store/dc3r2nbhckwn41hq5vaxxbj7z0lqdwb1-zlib-1.2.11/include\" ZLIB=\"-L/nix/store/dc3r2nbhckwn41hq5vaxxbj7z0lqdwb1-zlib-1.2.11/lib -lz\" UNIX=\"-DUNIX\"
       > make: Entering directory '/build/atasm109/src'
       > cc -g -Wall "-DZLIB_CAPABLE  "-DUNIX"  -c asm.c
       > /nix/store/dpjnjrqbgbm8a5wvi1hya01vd8wyvsq4-bash-4.4-p23/bin/bash: -c: line 0: unexpected EOF while looking for matching `"'
       > /nix/store/dpjnjrqbgbm8a5wvi1hya01vd8wyvsq4-bash-4.4-p23/bin/bash: -c: line 1: syntax error: unexpected end of file
       > make: *** [Makefile:29: asm.o] Error 1
       > make: Leaving directory '/build/atasm109/src'
       For full logs, run 'nix log /nix/store/az0zws725h9rk6paagw9dshxcbwjsr69-atasm-1.09.drv'.
```